File: core-rules/skills/anti-slop/profiles/go/fixtures/red.go
// Red fixture: every construct here must produce at least one anti-slop finding.
// One pattern per function, so inverting a rule breaks this file's expectation.
package fixtures

import (
	"encoding/json"
	"reflect"
)

// Widget is the named type the slop below refuses to commit to.
type Widget struct {
	Name string
}

// go-empty-interface: the contract accepts anything and proves nothing.
func Store(payload interface{}) error {
	return json.Unmarshal([]byte("{}"), &payload)
}

// go-empty-interface: `any` in signature position is the same discard, spelled newer.
func Widen(v any) string {
	return v.(Widget).Name
}

// go-unchecked-assert / forcetypeassert: panics instead of reporting the mismatch.
func WidgetName(raw any) string {
	return raw.(Widget).Name
}

// go-reflect: reads the shape at runtime instead of naming it in the signature.
func KindOf(payload any) string {
	return reflect.TypeOf(payload).Kind().String()
}

// errcheck: the marshalling error is dropped on the floor.
func StoreAndForget(w Widget) {
	json.Marshal(w)
}

File: core-rules/skills/anti-slop/profiles/rust/fixtures/red.rs
//! Red fixture: every construct here must produce at least one anti-slop finding.
//! One pattern per function, so inverting a lint breaks this file's expectation.

use std::num::NonZeroU32;

/// clippy::unwrap_used — throws the parse error away and panics in its place.
pub fn port_of(raw: &str) -> u16 {
    raw.parse().unwrap()
}

/// clippy::expect_used — a nicer panic message is still a panic.
pub fn retries_of(raw: &str) -> u32 {
    raw.parse().expect("retries must be a number")
}

/// clippy::as_conversions — silent truncation where a fallible convert belongs.
pub fn shard_index(total: u64) -> u32 {
    total as u32
}

/// clippy::undocumented_unsafe_blocks — no stated invariant for a reviewer to check.
pub fn nonzero(count: u32) -> NonZeroU32 {
    unsafe { NonZeroU32::new_unchecked(count) }
}
